Sources/Spezi/Configuration/Component.swift

```diff
@@ -47,19 +47,28 @@
 /// - ``ObservableObjectProvider``: A ``Component`` can conform to ``ObservableObjectProvider`` to inject `ObservableObject`s in the SwiftUI view hierarchy.
 ///
 /// All these protocols are combined in the ``Module`` protocol, making it an easy one-stop solution to support all these different functionalities and build a capable Spezi module.
-public protocol Component<ComponentStandard>: AnyObject, TypedCollectionKey {
+public protocol Component<ComponentStandard>: AnyObject, KnowledgeSource<SpeziAnchor> {
     /// A ``Component/ComponentStandard`` defines what ``Standard`` the component supports.
     associatedtype ComponentStandard: Standard
+
+    /// The `prepare` method is an optional method you may use to prepare your ``Component`` before it is configured (see ``configure()``).
+    ///
+    /// More specifically, the `prepare` method is called once the ``Component/StandardActor`` property is accessible and the ``Component/Provide``
+    /// properties are not yet collected.
+    func prepare()
```

Comment:
I was looking at the PR and the usage of the `prepare()` function and was wondering if the main reason behind using it compared to using the initializer is that we already have the standard accessible, allowing some decisions based on the standard provided and some communication with the standard instance?

Reply:
The problem that I would see is if the `Standard` is not yet fully configured at that stage, we would need to first enforce that all `Dependencies` of the Standard are configured? That would require some additional logic in the `Spezi` types. If we somehow enforce that all `@Dependencies` of the `Standard` are configured we will then have to ensure that they are called before and then we are going to have a race condition with the `prepare()` function or no real definition when it is called.

Given these thoughts, I would slightly favor removing the function and adding it at a point where we have a clear use case that goes beyond what we can do in the initializer?

Reply:
Agree, removed 👍 Thanks for the detailed explanation.